libraries/AP_HAL_SITL/UARTDriver.cpp

```diff
@@ -267,6 +267,7 @@ void UARTDriver::_tcp_start_connection(uint16_t port, bool wait_for_connection)
         }
         setsockopt(_fd, SOL_SOCKET, SO_REUSEADDR, &one, sizeof(one));
         setsockopt(_fd, IPPROTO_TCP, TCP_NODELAY, &one, sizeof(one));
+        fcntl(_fd, F_SETFD, FD_CLOEXEC);
```

Comment:
Why is this duplicated here?

Reply:
Can't have enough of them....

Which call is it duplicating?

Reply:
#L225 the socket should already being cloexeced !

Reply:
This isn't duplicated from line 255. This is a new socket and we can't rely on flag inheritance as not all OSes do the same.